Update Prometheus-adapter addon to version 0.6.0

Ebaneck opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->
'charts', 'prometheus-adapter', 

**Context**: 

See #2431 

**Summary**:

Bump Prometheus-adapter version to 0.6.0

**Acceptance criteria**: 

No regression

```
[root@bootstrap vagrant]# kubectl top node bootstrap
NAME        CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
bootstrap   346m         17%    2995Mi          81%     
```

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2431 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

Ebaneck committed:
charts: update prometheus-adapter to version 0.6.0

This chart is render using:

```
$ rm -rf charts/prometheus-adapter
$ helm repo update
$ helm fetch -d charts --untar stable/prometheus-adapter
$ ./charts/render.py prometheus-adapter --namespace metalk8s-monitoring charts/prometheus-adapter.yaml charts/prometheus-adapter/ > salt/metalk8s/addons/prometheus-adapter/deployed/chart.sls
```

Closes #2431

bert-e commented:
# Hello ebaneck,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2431-bump-prometheus-adapter-to-v0.6.0` into
`development/2.5`
* [w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote

TeddyAndrieux commented on salt/metalk8s/addons/prometheus-adapter/deployed/chart.sls:
```diff
@@ -1,55 +1,117 @@
 #!jinja | metalk8s_kubernetes
+
 {%- from "metalk8s/repo/macro.sls" import build_image_name with context %}
+{%- set prometheus_adapter = salt.metalk8s_service_configuration.get_service_conf('metalk8s-monitoring', 'unavailable') %}
```
I think this line should not be in this commit

Ebaneck committed:
renderer: remove mandatory `--service-config` argument

Some charts do not have a service config argument but the renderer
script enforces that we pass a service configuration.

This commit ensures the `--service-config` argument is not mandatory
while rendering the charts

bert-e commented:
# History mismatch



Merge commit #33bed9827107f1b495503db7205d043673eedf05 on the integration branch
`w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0` is merging a branch which is neither the current
branch `improvement/2431-bump-prometheus-adapter-to-v0.6.0` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2431-bump-prometheus-adapter-to-v0.6.0` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

Ebaneck commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2431-bump-prometheus-adapter-to-v0.6.0` into
`development/2.5`
* [w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote










*The following options are set:* **approve**

TeddyAndrieux commented on charts/render.py:
```diff
@@ -212,17 +213,20 @@ def main():
-        ).lstrip()
-    )
-    sys.stdout.write('\n')
+    if args.service_configs:
+        sys.stdout.write(
+            START_BLOCK.format(
+                configlines='\n'.join(
+                    ("{{%- set {} = salt.metalk8s_service_configuration"
+                     ".get_service_conf('{}', '{}') %}}").format(
+                        service_config[0], args.namespace, service_config[1]
+                    ) for service_config in args.service_configs
+                )
+            ).lstrip()
+        )
+        sys.stdout.write('\n')
+    else:
+        sys.stdout.write(START_BLOCK.format(configlines='').lstrip())
```
Would prefer an `if` only to set a `configlines` variable and then having the write with format

bert-e commented:
# History mismatch



Merge commit #d07a599b1a4265c52ce98bef021b47fe97ac9320 on the integration branch
`w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0` is merging a branch which is neither the current
branch `improvement/2431-bump-prometheus-adapter-to-v0.6.0` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2431-bump-prometheus-adapter-to-v0.6.0` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

Ebaneck commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2431-bump-prometheus-adapter-to-v0.6.0` into
`development/2.5`
* [w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2431-bump-prometheus-adapter-to-v0.6.0) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote










*The following options are set:* **approve**

TeddyAndrieux reviewed (approved):
Not tested but changes LGTM

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye ebaneck.
